Id/test/Id.test.cpp

```diff
@@ -0,0 +1,154 @@
+#define CATCH_CONFIG_RUNNER
+
+// other includes
+#include "catch.hpp"
+
+// what we're testing
+#include "utility/Id.hpp"
+
+int testNumber = 0;
+
+namespace {
+
+  // test variable
+  std::string str("myId");
+  std::string validstr("validId");
+
+  // dummy test class and validation function
+  class Dummy {};
+}
+
+// override validateId function for the Dummy class
+namespace utility {
+
+  template <> bool validateId<Dummy>(const std::string& id) noexcept {
+
+    if (id != validstr) {
+
+      return false;
+    }
+    return true;
```

Comment:
@jlconlin, This suggestion is not the convention used elsewhere in the NJOY21 source code.  

Requiring an `else` block to follow an `if` statement with a return statement just introduces an additional level of indentation.

In general, my preference for these sorts of if-return-else-return blocks is a ternary statement, e.g.
```
return ( id != validstr) ? false : true;
```
but given this particular case, in opinion
```
return (id == validstr);
```
is the least noisy.